src/ucs/datastruct/mpmc.c

```diff
@@ -17,9 +17,9 @@
 #include <ucs/debug/memtrack.h>
 
 
-ucs_status_t ucs_mpmc_queue_init(ucs_mpmc_queue_t *mpmc, uint32_t length)
+ucs_status_t ucs_mpmc_queue_init(ucs_mpmc_queue_t *mpmc, uint64_t length)
```

Comment:
this PR does not really require changing the size to u64, but only the value.

Reply:
UCS_MPMC_VALID_SHIFT should be changed to 63 so that the assert here would pass - https://github.com/openucx/ucx/blob/master/src/ucs/datastruct/mpmc.c#L58 , and this param is also used here, for example, https://github.com/openucx/ucx/blob/master/src/ucs/datastruct/mpmc.c#L26 , which won't be correct if length is uint32_t .
is this a problem having all of them 64 bits?

Reply:
https://github.com/openucx/ucx/blob/master/src/ucs/datastruct/mpmc.c#L26 should work also with length is 32bit
in general it's not a problem but if already changing then length should be size_t and ther eshould not be a hard coupling between length and element size

Reply:
@yosefe coverity fails on this line after restoring length to 32bits - 
https://github.com/openucx/ucx/blob/master/src/ucs/datastruct/mpmc.c#L26
since it's always false now.
make length 64bits then? size_t
or change this check not to use UCS_MPMC_VALID_SHIFT? which is 63 now

Reply:
try to cast length to uint64_t in this check

Reply:
but it wont change the value of length...

Reply:
@yosefe (forgot to tag)

Reply:
yes, but maybe will silence the warning

Reply:
but then we don't need this 'if' at all if we always know the answer to it..

Reply:
so make it a static assert?

Reply:
@yosefe how about:
```
if (mpmc->length >= UCS_BIT(31)) {
     return UCS_ERR_INVALID_PARAM;
}
```
the way it was before the change of making the data 64bit.

Reply:
@yosefe cannot use a static assert since mpmc->length isn't a const and depends on the 'length' param.
`
error: case label does not reduce to an integer constant
`

casting mpmc->length to uint64_t doesn't work either:

`
(1) Event result_independent_of_operands: 	"(uint64_t)mpmc->length >= (9223372036854775808UL /* 1UL << 63 */)" is always false regardless of the values of its operands. This occurs as the logical operand of "if".
`

Reply:
i see, can you suggest some other way to check it?

Reply:
i've been trying several ways. with casting and using new params and passing to a new function that would accept length as a const but nothing worked so far.